lib/sycamore/sycamore/transforms/extract_graph_relationships.py

```diff
@@ -120,9 +120,12 @@ async def _generate_relationships(self, section: HierarchicalDocument) -> dict:
 
             if start_nodes and end_nodes:
                 parsed_relations.append(relation)
-                parsed_metadata[relation.__name__] = {"start_label": start_label, "end_label": end_label}
-                parsed_nodes[start_label] |= set(start_nodes)
-                parsed_nodes[end_label] |= set(end_nodes)
+                parsed_metadata[relation.__name__] = {
+                    "start_label": start_label,
+                    "end_label": end_label,
+                    "start_nodes": set(start_nodes),
+                    "end_nodes": set(end_nodes),
```

Comment:
wait are these really supposed to be sets? sets are unordered and stuff. what if a relation is not fully-connected?

Reply:
yeah so this is not a big worry, its essentially just part of building context for the LLM. This stuff is passed to the prompt to generate it. I'm also going to be reworking logic around this area in the next PR. Here we create a set to of start nodes and end nodes and then pass them to the LLM as enums so it chooses a node that we have already extracted instead of hallucinating something.

Reply:
oh right. You explained that to me at some point my bad